FenixApplication/app/src/main/java/com/example/fenixapplication/db/SQLiteHelper.kt

```diff
@@ -71,4 +107,52 @@ class SQLiteHelper(context: Context?) : SQLiteOpenHelper(context, DATABASE_NAME,
         return stdList
     }
 
+    // TODO - function need be working when get connection
+    @SuppressLint("Range")
+    fun getDeviceStatusHistory(deviceID: Int): ArrayList<StatusModel> {
+        val stdList: ArrayList<StatusModel> = ArrayList()
+        val selectQuery = "SELECT status_date, status_min, status_current, status_max FROM $TBL_STATUS WHERE status_to_device = $deviceID"
+        val db = this.readableDatabase
+        val cursor: Cursor?
+
+        try {
+            cursor = db.rawQuery(selectQuery, null)
+        } catch (e: Exception) {
+            e.printStackTrace()
+            db.execSQL(selectQuery)
+            return ArrayList()
+        }
+
+        var statusId: Int
+        var statusToDevice: String
+        var statusDate: String
+        var statusMin: String
+        var statusCurrent: String
+        var statusMax: String
+
+        if(cursor.moveToFirst()) {
+            do {
+                statusId = cursor.getInt(cursor.getColumnIndex("status_id"))
+                statusToDevice = cursor.getString(cursor.getColumnIndex("status_to_device"))
+                statusDate = cursor.getString(cursor.getColumnIndex("status_date"))
+                statusMin = cursor.getString(cursor.getColumnIndex("status_min"))
+                statusCurrent = cursor.getString(cursor.getColumnIndex("status_current"))
+                statusMax = cursor.getString(cursor.getColumnIndex("status_max"))
+                val std = StatusModel(status_id = statusId, status_to_device = statusToDevice,
+                    status_date = statusDate, status_min = statusMin, status_current = statusCurrent, status_max = statusMax)
+                stdList.add(std)
+
+            } while (cursor.moveToFirst())
```

Comment:
maybe for clarity close do in line 144 and only while loop in line 145
do {
stuff here
}
while()